dll/win32/kernel32/client/proc.c

```diff
@@ -123,13 +123,10 @@ WINAPI
 BasepIsImageVersionOk(IN ULONG ImageMajorVersion,
                       IN ULONG ImageMinorVersion)
 {
-    /* Accept images for NT 3.1 or higher, as long as they're not newer than us */
+    /* Accept images for NT 3.1 or higher */
     return ((ImageMajorVersion >= 3) &&
             ((ImageMajorVersion != 3) ||
-             (ImageMinorVersion >= 10)) &&
-            (ImageMajorVersion <= SharedUserData->NtMajorVersion) &&
-            ((ImageMajorVersion != SharedUserData->NtMajorVersion) ||
-             (ImageMinorVersion <= SharedUserData->NtMinorVersion)));
+             (ImageMinorVersion >= 10)));
```

Comment:
I'm just nitpicking, but that check is extremely confusing on first read. I'd change it to
```c
     return ((ImageMajorVersion > 3) ||
             ((ImageMajorVersion == 3) && (ImageMinorVersion >= 10)));
```

Reply:
I would keep the original code that is *correct* commented out, with a comment mentioning why it has been replaced.

Reply:
Hmmm actually I think that this would be better done with a shim, that modifies for the given process the values of the NtMajorVersion / NtMinorVersion members ?

Reply:
Why keep old code in a comment? That's what the git history is for! Let's not keep random code blocks commented out, please. :P

Reply:
Either ifdef with some define that can be disabled (if someone wants to), or a comment above it, that this is reactos specific behavior.